src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -341,13 +341,21 @@ private void GenerateFile(List<string> typeNames, string assemblyName, bool prox
         // assumes all same case.        
         private bool ArgumentMatch(string arg, string formal)
         {
-            if (arg[0] != '/' && arg[0] != '-')
+            if (arg[0] != '-' )
             {
                 return false;
             }
 
+            // Full name format, eg: --assembly
+            if (arg.Length > 1 && arg[1] == '-')
+            {
+                arg = arg.Substring(2);
+                return arg == formal;
+            }
+
+            // Short name format, eg: -a
             arg = arg.Substring(1);
-            return (arg == formal || (arg.Length == 1 && arg[0] == formal[0]));
+            return arg.Length == 1 && arg[0] == formal[0];
```

Comment:
This can be problematic. For example, if we support two arguments, assembly and additionalparam, -a will match whichever we compare first. Sometimes, a short form can be more than one letter too. We should only support the short form arguments that we meant to support.